hudi-common/src/main/java/org/apache/hudi/common/util/ClosableIterator.java

```diff
@@ -24,8 +24,29 @@
  * An iterator that give a chance to release resources.
  *
  * @param <R> The return type
+ *
+ * TODO move under common.util.collection
  */
 public interface ClosableIterator<R> extends Iterator<R>, AutoCloseable {
   @Override
   void close(); // override to not throw exception
+
+  static <E> ClosableIterator<E> wrap(Iterator<E> iterator) {
+    return new ClosableIterator<E>() {
+      @Override
+      public boolean hasNext() {
```

Comment:
Does the wrap make any sense here?

Reply:
So the idea is that we can wrap any Iterator and make it Closeable.
Where do you suggest this should be placed?